ArduCopter/land_detector.cpp

```diff
@@ -196,7 +196,7 @@ void Copter::update_throttle_mix()
         // check if landing
         const bool landing = flightmode->is_landing();
 
-        if ((large_angle_request && !landing) || large_angle_error || accel_moving || descent_not_demanded) {
+        if (((large_angle_request || force_flying) && !landing) || large_angle_error || accel_moving || descent_not_demanded) {
```

Comment:
I guess I don't understand this logic - conceptually it seems to me that you want to force flying to prevent you otherwise landing. So based on the names of the variables I would expect something like:
```
if ((large_angle_request && !landing) || force_flying || large_angle_error || accel_moving || descent_not_demanded) {
```
but I am clearly not understanding the logic

Reply:
We want this to force flying in all conditions EXCEPT when the aircraft is attempting to land in an auto mode. So we prevent landing detection (through drop to min throttle) unless the aircraft is in the final stage of landing.